cpp/include/graph_generators.hpp

```diff
@@ -0,0 +1,314 @@
+                                size_t x,
+                                size_t y,
+                                size_t num_graphs,
+                                bool symmetrize = false);
+
+/**
+ * @brief generate an edge list for a 3D Mesh Graph
+ *
+ * A 3D mesh graph will be constructed with dimension @p x by @p y by @p z.
+ * @p num_graphs of this size will be constructed
+ *
+ * If executed in a multi-gpu context (handle comms has been initialized)
+ * each GPU will generate disjoint mesh constructs of equal size.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param x Number of vertices in X dimension
+ * @param y Number of vertices in Y dimension
+ * @param z Number of vertices in Z dimension
+ * @param num_graphs Number of mesh graphs to generate
+ * @param symmetrize If true, symmetrize the edges
+ * @return std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>> A tuple of
+ * rmm::device_uvector objects for edge source vertex IDs and edge destination vertex IDs.
+ */
+template <typename vertex_t>
+std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>>
+generate_3d_mesh_graph_edgelist(raft::handle_t const& handle,
+                                size_t x,
+                                size_t y,
+                                size_t z,
+                                size_t num_graphs,
+                                bool symmetrize = false);
+
+/**
+ * @brief generate an edge lists for some complete graphs
+ *
+ * A collection of @p num_graphs complete graphs, each containing
+ * @p num_vertices is generated.
+ *
+ * If executed in a multi-gpu context (handle comms has been initialized)
+ * each GPU will generate disjoint complete graph constructs of equal size.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param num_vertices Number of vertices to use in each complete graph
+ * @param num_graphs Number of graphs to generate
+ * @param symmetrize If true, symmetrize the edges
+ * @return std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>> A tuple of
+ * rmm::device_uvector objects for edge source vertex IDs and edge destination vertex IDs.
+ */
+template <typename vertex_t>
+std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>>
+generate_complete_graph_edgelist(raft::handle_t const& handle,
+                                 size_t num_vertices,
+                                 size_t num_graphs,
+                                 bool symmetrize = false);
+
+/**
+ * @brief generate an edge lists for an Erdos-Renyi graph
```

Comment:
Better explicitly say you're supporting the G(n, p) model.

Note that the G(n, p) approach is O(V^2); you don't need to worry about handling duplicates in contrast to G(n, m) but generating a graph with large n and small p becomes prohibitively expensive.

And if we allow small number of duplicates, R-mat graph with a=b=c=d is a good approximation of the ER random graph.